src/ReflectionFunctionAbstract.php

```diff
@@ -48,6 +48,22 @@ protected function populateFunctionAbstract(MethodNode $node, $filename)
             $this->docBlock = $comment->getReformattedText();
         }
 
+        // We must determine if params are optional or not ahead of time, but
+        // we must do it in reverse...
+        $overallOptionalFlag = true;
+        for ($i = count($node->params) - 1; $i >= 0; $i--) {
+            $hasDefault = !is_null($node->params[$i]->default);
+
+            // When we find the first parameter that does not have a default,
+            // flip the flag as all params for this are no longer optional
+            // EVEN if they have a default value
```

Comment:
Yes it's stupid, but it accounts for situations like this:

```php
function myFunc($foo = 'foo', $bar)
```

In this example, `isOptional()` on the `$foo` should return `false`, but `isDefaultValueAvailable()` should return `true`. AST doesn't give us information about whether params are "optional" or not, so we need to start with the last parameter, and work backwards, determining if each is optional.